### ISSUE TITLE
hs.application.get regression in v0.9.98 (6805), given same input now yields different results / nil

### ISSUE THREAD

**Issue by viddo** [Hammerspoon/hammerspoon#3423]
Noticed this just after updating today and a script of mine stopped working.

In my case and using a concrete example: I run both [VSCode](https://code.visualstudio.com/) and it's latest build version aka [VSCode insiders](https://code.visualstudio.com/insiders/) simultaneously, but now it appears I can only get a hold of the _insider_ app using this API. I've used `hs.application.get('Code')` and `hs.application.get('Code - Insiders')` respectively (screenshots from quick sanity testing in console):
| Prev version: 0.9.97 (6267) | Latest version 0.9.98 (6805) _at time of writing_ |
| - | - | 
| <img width="403" alt="Hammerspoon 0 9 97 (6267)" src="https://user-images.githubusercontent.com/978461/224690633-4da39e75-dc28-48ed-9cfb-73ceeab0f715.png"> |  <img width="444" alt="Hammerspoon 0 9 98 (6805)" src="https://user-images.githubusercontent.com/978461/224690663-0f58039f-af5c-46ca-a0d5-c4eb9bccf5da.png"> |

Didn't bother testing other windows but suspect there might be similar issues with other apps?

FWIW, they are listed under `runningApplications` with same previous names:
```lua
for _, v in pairs(hs.application.runningApplications()) do print(v) end
…
2023-03-13 12:42:34: hs.application: Code - Insiders (0x600001010af8)
2023-03-13 12:42:34: hs.application: Code (0x6000010131f8)
```

**Comment by cmsj** [comment]
I'm curious what would happen if your `runningApplications()` loop there was `print(v:name())` instead, which is what `hs.application.find()` is checking under the hood in this codepath.

**Comment by cmsj** [comment]
Never mind, I installed Code - Insiders and reproduced the issue - turns out it's the `-` being interpreted as a Lua pattern matching operator. I believe I have resolved this in #3425 and will merge that soon. I need to do a release imminently to address some other issues in 0.9.98, and this will be included.

**Comment by viddo** [comment]
That was fast, thanks for the quick turnaround and update! 🙇👌
